firefox-ios/Client/Frontend/Browser/Tabs/Middleware/TabManagerMiddleware.swift

```diff
@@ -392,6 +392,10 @@ class TabManagerMiddleware: BookmarksRefactorFeatureFlagProvider {
                                                        windowUUID: uuid,
                                                        actionType: GeneralBrowserActionType.showToast)
                 store.dispatch(toastAction)
+                let tabManager = tabManager(for: uuid)
```

Comment:
We could add a function for the logic to add a new regular tab that is repeated here and in `closeAllTabs. ' I think we should also add a comment to clarify the case we are handling because it's an edge case not obvious. I remember in the old implementation, we have these comments to help understand it